src/intl/context.js

```diff
@@ -1,16 +1,22 @@
-import FTLRuntimeParser from '../ftl/entries/parser';
 import { format } from './resolver';
 
+const { classes: Cc, interfaces: Ci, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "L20nParser",
+                                  "resource://gre/modules/L20nParser.jsm");
+
 export class MessageContext {
-  constructor(lang, { functions = {} } = {}) {
-    this.lang = lang;
+  constructor(locales, { functions = {}, args = {} } = {}) {
+    this.locale = Array.isArray(locales) ? locales[0] : locales;;
```

Comment:
I doubt we'll be able to push MessageContext for standardization with other signature than "locales, options". And if we don't want to push for standardization then maybe we should not put it in Intl?

I do believe that we'll need some (public) API to negotiate locales for multiple formatters - like give me the best locale fallback chain for Messages/Units/DateTime. And we should aim to polyfill that.